client/pages/topsecret.jsx

```diff
@@ -13,31 +63,125 @@ const HomePage = React.createClass({
       console.log(this.context);
     }
 
+
   },
 
   render() {
+    let initialBankSavings = this.state.initialBankSavings;
+    let initialExpenses = this.state.initialExpenses;
+    let tools = this.state.initialBankSavings;
+    let addlbl = this.state.addlbl;
+    let skuin = this.state.skuin;
+    let typein = this.state.typein;
+    let namein = this.state.namein;
+    let costin = this.state.costin;
+    let nameQ = this.state.nameQ;
+    let costQ = this.state.costQ;
+    let searchdebug = this.state.searchdebug;
+    let costfilter = this.state.costfilter;
+
+    const renderedArmory = _.map(this.state.tools, (tool) => {
+      if(this.state.tools.length == 0){
+        return null;
+      }
+      return (
+        <ReactBS.ListGroupItem> SKU: {tool.sku}, Type: {tool.type}, Name: {tool.name}, Cost: ${tool.cost}</ReactBS.ListGroupItem>
+      )
+    }
+    );
+
+    const renderedNameSearch = _.map(this.state.tools, (tool) => {
+      if(this.state.nameQ == '' && this.state.costQ == ''){
+        return null;
+      }
+      else if(this.state.costQ == '' && tool.name.indexOf(this.state.nameQ) > -1){
+        return (<ReactBS.ListGroupItem> SKU: {tool.sku}, Type: {tool.type}, Name: {tool.name}, Cost: ${tool.cost}</ReactBS.ListGroupItem>);
+      }
+      else if(tool.name.indexOf(this.state.nameQ) > -1 && this.state.costQ != '' && this.state.costfilter == "=" && tool.cost === Number(this.state.costQ)){ //
+        return (<ReactBS.ListGroupItem> SKU: {tool.sku}, Type: {tool.type}, Name: {tool.name}, Cost: ${tool.cost}</ReactBS.ListGroupItem>);
+      }
+      else if(this.state.costQ != '' && this.state.costfilter == ">" && tool.cost > Number(this.state.costQ)){
+        if(this.state.costQ == ''){
+          return (<ReactBS.ListGroupItem> SKU: {tool.sku}, Type: {tool.type}, Name: {tool.name}, Cost: ${tool.cost}</ReactBS.ListGroupItem>);
+        }
+        else {
+          if(tool.name.indexOf(this.state.nameQ) > -1) return (<ReactBS.ListGroupItem> SKU: {tool.sku}, Type: {tool.type}, Name: {tool.name}, Cost: ${tool.cost}</ReactBS.ListGroupItem>);
+        }
+      }
+      else if(this.state.costQ != '' && this.state.costfilter == "<" && tool.cost < Number(this.state.costQ)){
+        if(this.state.costQ == ''){
+          return (<ReactBS.ListGroupItem> SKU: {tool.sku}, Type: {tool.type}, Name: {tool.name}, Cost: ${tool.cost}</ReactBS.ListGroupItem>);
+        }
+        else {
+          if(tool.name.indexOf(this.state.nameQ) > -1) return (<ReactBS.ListGroupItem> SKU: {tool.sku}, Type: {tool.type}, Name: {tool.name}, Cost: ${tool.cost}</ReactBS.ListGroupItem>);
+        }
+      }
+      else if(this.state.costQ != '' && this.state.costfilter == "=" && tool.cost === Number(this.state.costQ)){
+        if(this.state.costQ == ''){
+          return (<ReactBS.ListGroupItem> SKU: {tool.sku}, Type: {tool.type}, Name: {tool.name}, Cost: ${tool.cost}</ReactBS.ListGroupItem>);
```

Comment:
It looks like the returned JSX in this if statement is the same JSX returned by the else.